deeppavlov/skills/aiml_skill/aiml_skill.py

```diff
@@ -0,0 +1,139 @@
+import uuid
+from pathlib import Path
+from typing import Union
+from typing import Tuple, Optional, List
+from logging import getLogger
+import aiml
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.skill.skill import Skill
+
+log = getLogger(__name__)
+
+
+@register("aiml_skill")
+class AIMLSkill(Skill):
+    """Skill wraps python-aiml library into DeepPavlov interfrace.
+    AIML uses directory with AIML scripts which are loaded at initialization and used as patterns
+    for answering at each step.
+    """
+
+    def __init__(self,
+                 path_to_aiml_scripts: str,
+                 positive_confidence: float = 0.66,
+                 null_response: str = "I don't know what to answer you",
+                 null_confidence: float = 0.33,
+                 **kwargs
+                 ) -> None:
+        """
+        Construct skill:
+            read AIML scripts,
+            load AIML kernel
+
+        Args:
+            path_to_aiml_scripts: string path to folder with AIML scripts
+            null_response: Response string to answer if no AIML Patterns matched
+            positive_confidence: The confidence of response if response was found in AIML scripts
+            null_confidence: The confidence when AIML scripts has no rule for responding and system returns null_response
+        """
+        # we need absolute path (expanded for user home and resolved if it relative path):
+        self.path_to_aiml_scripts = Path(path_to_aiml_scripts).expanduser().resolve()
+        log.info(f"path_to_aiml_scripts is: `{self.path_to_aiml_scripts}`")
+
+        self.positive_confidence = positive_confidence
+        self.null_confidence = null_confidence
+        self.null_response = null_response
+        self.kernel = aiml.Kernel()
+        # to block AIML output:
+        self.kernel._verboseMode = False
+        self._load_scripts()
+
+    def _load_scripts(self) -> None:
+        """
+        Scripts are loaded recursively from files with extensions .xml and .aiml
+        Returns: None
+
+        """
+        # learn kernel to all aimls in directory tree:
+        all_files = sorted(self.path_to_aiml_scripts.rglob('*.*'))
+        learned_files = []
+        for each_file_path in all_files:
+            if each_file_path.suffix in ['.aiml', '.xml']:
+                # learn the script file
+                self.kernel.learn(str(each_file_path))
+                learned_files.append(each_file_path)
+        if not learned_files:
+            log.error(f"No .aiml or .xml files found for AIML Kernel in directory {self.path_to_aiml_scripts}")
```

Comment:
You should `raise` if it's an error.
Otherwise you can `log.warning()` instead of `log.error()`

Reply:
fixed